web/src/main/java/org/openmrs/web/filter/initialization/InitializationFilter.java

```diff
@@ -1492,7 +1516,7 @@ public void run() {
 							runtimeProperties.put("connection.driver_class", wizardModel.databaseDriver);
 						}
 						if (finalDatabaseConnectionString.contains("postgres")) {
-							runtimeProperties.put("hibernate.dialect", "org.hibernate.dialect.PostgreSQLDialect");
```

Comment:
Could you remind me as to why we are using PostgreSQL82Dialect instead of PostgreSQL92Dialect or PostgreSQL102Dialect, etc?

Reply:
PostgreSQL82Dialect supports PostgreSQL 8.2 and later ( see here : https://docs.jboss.org/hibernate/orm/5.1/javadocs/org/hibernate/dialect/PostgreSQL82Dialect.html ) where as others like PostgreSQL10 Dialect provide support for PostgreSQL10 and later ( see here: https://docs.jboss.org/hibernate/stable/orm/javadocs/org/hibernate/dialect/PostgreSQL10Dialect.html ). I used PostgreSQL82Dialect so that if some implementation is still relying on older versions of PostgreSQL then there won't be a problem.

Reply:
Does this therefore mean that we do not support the latest release of PostgreSQL?

Reply:
I would actually questions whether we need to select a Hibernate dialect at all, but that might be bigger than this ticket (Hibernate should automatically [do the right thing](https://docs.jboss.org/hibernate/stable/orm/userguide/html_single/Hibernate_User_Guide.html#portability-dialectresolver)).

Reply:
> Does this therefore mean that we do not support the latest release of PostgreSQL?

Here as mentioned in the docs, support is for PostgreSQL 8.2 and later.
https://docs.jboss.org/hibernate/orm/5.1/javadocs/org/hibernate/dialect/PostgreSQL82Dialect.html

Reply:
What do you think about @ibacher's point?

Reply:
> I would actually questions whether we need to select a Hibernate dialect at all, but that might be bigger than this ticket (Hibernate should automatically [do the right thing](https://docs.jboss.org/hibernate/stable/orm/userguide/html_single/Hibernate_User_Guide.html#portability-dialectresolver)).

@ibacher your take on this appears to me, right. I'll try on mysql and postgresql without specifying the dialect and see what happens. Though, I fear if there is a problem then it might require a separate ticket in itself and also a lot of look into how hibernate uses database metadata( like using drivers ) to select dialect  and won't be of immediate concern.

Reply:
Can you try it out and tell us what happens?

Reply:
Sure @dkayiwa , I am working on it.

Reply:
@ibacher  @dkayiwa  , I initialized OpenMRS-core on PostgreSQL. I added legacyUI module and Now when I am trying to create a Patient, 

1. Without using any PostgreSQL dialect : This is the error log https://pastebin.com/JaCZdja6  

2. Using PostgreSQL dialect: Everything works fine and I am redirected to page where I need to fill patient details.

Reply:
Did you test to confirm that `PostgreSQL82Dialect` works for the latest release of PostgreSQL?

Reply:
> @ibacher @dkayiwa , I initialized OpenMRS-core on PostgreSQL. I added legacyUI module and Now when I am trying to create a Patient,
> 
>     1. Without using any PostgreSQL dialect : This is the error log [pastebin.com/JaCZdja6](https://pastebin.com/JaCZdja6)
> 
>     2. Using PostgreSQL dialect: Everything works fine and I am redirected to page where I need to fill patient details.

@Am-Coder I think there may be a small trick to this. There's a file in `src/main/resources` called `hibernate.default.properties` which has the line:

```
hibernate.dialect=org.hibernate.dialect.MySQLDialect
```

Could you edit that file to comment out that line and try again?

Reply:
> > @ibacher @dkayiwa , I initialized OpenMRS-core on PostgreSQL. I added legacyUI module and Now when I am trying to create a Patient,
> > ```
> > 1. Without using any PostgreSQL dialect : This is the error log [pastebin.com/JaCZdja6](https://pastebin.com/JaCZdja6)
> > 
> > 2. Using PostgreSQL dialect: Everything works fine and I am redirected to page where I need to fill patient details.
> > ```
> 
> @Am-Coder I think there may be a small trick to this. There's a file in `src/main/resources` called `hibernate.default.properties` which has the line:
> 
> ```
> hibernate.dialect=org.hibernate.dialect.MySQLDialect
> ```
> 
> Could you edit that file to comment out that line and try again?

Sorry, I missed the default dialect. So, I tried and the results are:

1. In both mysql and postgresql we are able enter Patient details page successfully. In mysql we are able to add patient successfully as well.
2. In mysql, I entered patient management section and tried to delete the patient . The error log I got is : https://pastebin.com/cj0tb7ue . I think this might be related to this that https://talk.openmrs.org/t/patient-management-in-postgresql/28228/5 .

Reply:
> Did you test to confirm that `PostgreSQL82Dialect` works for the latest release of PostgreSQL?

I'll try with postgresql 12 and report the results.

Reply:
@Am-Coder Does 1 imply that you weren't able to add a patient successfully on PostgreSQL? 2 is a very good point; I forgot we were being "saved" by using the "wrong" dialect. I'm actually in favour of just adding in the appropriate `ondelete = cascade` properties to the affected tables.

Reply:
@ibacher  
For 1 there is another problem that is not related to dialect but related to sequence and holds a different solution.

For 2, I will be adding the necessary ondelete = cascade in another ticket as this is necessary for PostgreSQL.

Reply:
@Am-Coder didn't i solve the sequence problem in my previous commits?

Reply:
> @Am-Coder didn't i solve the sequence problem in my previous commits?

Still the problem exists, the error log : https://pastebin.com/VJrSf98T